### DISCUSSION TITLE
Splited go version in Dockerfile

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#21764] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

Gitlab EE

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

A snip of the Dockerfile:
```dockerfile
ENV NAME=golang \
    GO_MAJOR_VERSION=1 \
    GO_MINOR_VERSION=19 \
    GO_PATCH_VERSION=4 

ENV VERSION=$GO_MAJOR_VERSION.$GO_MINOR_VERSION.$GO_PATCH_VERSION

```
I want to control which part of the go version to upgrade. Sometimes it would be only patches, sometimes, minor versions and sometimes major versions.

For example: Patch upgrading - Let's assume there is a new go patch `1.19.5`.  Renovate should produce:
```dockerfile
ENV NAME=golang \
    GO_MAJOR_VERSION=1 \
    GO_MINOR_VERSION=19 \
    GO_PATCH_VERSION=5

ENV VERSION=$GO_MAJOR_VERSION.$GO_MINOR_VERSION.$GO_PATCH_VERSION

```

Minor upgrading - A new go minor version `1.20.3`. Renovate should produce:
```dockerfile
ENV NAME=golang \
    GO_MAJOR_VERSION=1 \
    GO_MINOR_VERSION=20 \
    GO_PATCH_VERSION=3

ENV VERSION=$GO_MAJOR_VERSION.$GO_MINOR_VERSION.$GO_PATCH_VERSION

```

### What you tried so far.

I have no idea how to approach this use case

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Maybe regex manager and regex versioning? Capture the 3 lines as the currentValue, then use regex versioning to extract our major, minor, patch

A couple of notes:
- I think that the `\` style chosen is a little quirky, could also be separate `ENV` lines
- Careful that Go doesn't technically have a `.0` new release, e.g. `1.20` means `1.20.0` _except_ for Docker images which do have `1.20.0`. They are considering changing this in future
